tests/CatalogTests/CatalogIndexReaderTests.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using NgTests.Data;
+using NuGet.Services.Metadata.Catalog;
+using Xunit;
+
+namespace CatalogTests
+{
+    public class CatalogIndexReaderTests
+    {
+        [Fact]
+        public async Task GetEntries()
+        {
+            // Arrange
+            var indexUri = "http://tempuri.org/index.json";
+            var responses = new Dictionary<string, string>()
+            {
+                { "http://tempuri.org/index.json", TestCatalogEntries.TestCatalogStorageWithThreePackagesIndex },
+                { "http://tempuri.org/page0.json", TestCatalogEntries.TestCatalogStorageWithThreePackagesPage },
+            };
+
+            var reader = new CatalogIndexReader(
+                new Uri(indexUri),
+                new CollectorHttpClient(new InMemoryHttpHandler(responses)));
+
+            // Act
+            var entries = await reader.GetEntries();
+
+            // Assert
+            var entryList = entries.ToList();
+            Assert.Equal(3, entryList.Count);
+
+            Assert.Equal("http://tempuri.org/data/2015.10.12.10.08.55/listedpackage.1.0.1.json", entryList[0].Uri.ToString());
+            Assert.Equal("http://tempuri.org/data/2015.10.12.10.08.54/listedpackage.1.0.0.json", entryList[1].Uri.ToString());
+            Assert.Equal("http://tempuri.org/data/2015.10.12.10.08.54/unlistedpackage.1.0.0.json", entryList[2].Uri.ToString());
+
+            Assert.Equal("2015-10-12T10:08:55.3335317", entryList[0].CommitTimeStamp.ToString("O"));
+            Assert.Equal("2015-10-12T10:08:54.1506742", entryList[1].CommitTimeStamp.ToString("O"));
+            Assert.Equal("2015-10-12T10:08:54.1506742", entryList[2].CommitTimeStamp.ToString("O"));
+
+            Assert.Same(entryList[0].Id, entryList[1].Id);
```

Comment:
Check the value directly as well? If `entryList[0].Id` and `entryList[1].Id` are both null or some equally wrong value this won't catch it.
```
Assert.Equal("listedpackage", entryList[0].Id);
Assert.Equal("listedpackage", entryList[1].Id);
Assert.Equal("unlistedpackage", entryList[2].Id);
```